src/components/VotePanel.tsx

```diff
@@ -14,21 +15,24 @@ export default function VotePanel(_props: Props) {
   const { data } = usePublishedDatum(
     `wallet.${walletUtils.getWalletAddress()}.current`
   );
+  const { status: instanceStatus, data: instance } = usePublishedDatum(
+    'agoricNames.instance'
+  );
+
+  let eligibility = <p>Loading…</p>;
```

Comment:
please factor out an `Eligibility` component like in PsmPanel so there's no local state for it and it can be expressed as a `switch`. The switch is better than cascading if because every condition is a case of a single value.

Reply:
done